tests/checkpointing/test_model_checkpoint.py

```diff
@@ -769,11 +768,21 @@ def test_default_checkpoint_behavior(tmpdir):
         default_root_dir=tmpdir, max_epochs=3, progress_bar_refresh_rate=0, limit_train_batches=5, limit_val_batches=5
     )
 
-    trainer.fit(model)
-    results = trainer.test()
+    with patch.object(ModelCheckpoint, "_save_model", wraps=trainer.checkpoint_callback._save_model) as save_mock:
+        trainer.fit(model)
+        results = trainer.test()
 
     assert len(results) == 1
-    assert len(trainer.dev_debugger.checkpoint_callback_history) == 3
+    assert save_mock.call_count == 3
+
+    save_dir = tmpdir / "lightning_logs" / "version_0" / "checkpoints"
+    save_mock.assert_has_calls(
+        [
+            call(trainer, save_dir / "epoch=0-step=4.ckpt"),
+            call(trainer, save_dir / "epoch=1-step=9.ckpt"),
+            call(trainer, save_dir / "epoch=2-step=14.ckpt"),
+        ]
+    )
```

Comment:
since you added this, remove the lines below?

```diff
-    # make sure the checkpoint we saved has the metric in the name
-    ckpts = os.listdir(os.path.join(tmpdir, "lightning_logs", "version_0", "checkpoints"))
-    assert len(ckpts) == 1
-    assert ckpts[0] == "epoch=2-step=14.ckpt"
```

Reply:
I think we still want to assert that only the the latest one is saved and that the others are not present on the filesystem.